src/inversion_scripts/operators/TROPOMI_operator.py

```diff
@@ -143,31 +160,107 @@ def apply_average_tropomi_operator(
+            xch4 = {}
+                
+            for v in vars_to_xch4:
+                # Get GEOS-Chem jacobian ch4 at this lat/lon, for all vertical levels and state vector elements
+                jacobian_lonlat = GEOSCHEM[v][
+                    gridcell_dict["iGC"], gridcell_dict["jGC"], :, :
+                ]
+                # Map the sensitivities to TROPOMI pressure levels
+                sat_deltaCH4 = remap_sensitivities(
+                    jacobian_lonlat,
+                    merged["data_type"],
+                    merged["p_merge"],
+                    merged["edge_index"],
+                    merged["first_gc_edge"],
+                )  # mixing ratio, unitless
+                # Tile the TROPOMI averaging kernel
+                avkern_tiled = np.transpose(np.tile(avkern, (n_elements, 1)))
+                # Tile the TROPOMI dry air subcolumns
+                dry_air_subcolumns_tiled = np.transpose(
+                    np.tile(dry_air_subcolumns, (n_elements, 1))
+                )  # mol m-2
+                # Derive the change in column-averaged XCH4 that TROPOMI would see over this ground cell
+                #jacobian_K[i, :] = np.sum(
+                xch4[v] = np.sum(
+                    avkern_tiled * sat_deltaCH4 * dry_air_subcolumns_tiled, 0
+                ) / sum(
+                    dry_air_subcolumns
+                )  # mixing ratio, unitless
+            
+            # separate variables for convenience later
+            jacobian_xch4 = xch4['jacobian_ch4']
+            emis_base_xch4 = xch4['emis_base_ch4']
+            if config['OptimizeOH']:
+                oh_base_xch4 = xch4['oh_base_ch4']
+            
+            
+            # Calculate sensitivities and save in K matrix
+            # determine which elements are for emis,
+            # BCs, and OH
+            is_oh = np.full(n_elements, False, dtype=bool)
+            is_bc = np.full(n_elements, False, dtype=bool)
+            is_emis = np.full(n_elements, False, dtype=bool)
+            
+            for e in range(n_elements):
+                i_elem = e + 1
+                # booleans for whether this element is a
+                # BC element or OH element
+                is_OH_element = check_is_OH_element(
+                    i_elem, n_elements, config['OptimizeOH']
+                )
 
+                is_BC_element = check_is_BC_element(
+                    i_elem, n_elements, config['OptimizeOH'],
+                    config['OptimizeBCs'], is_OH_element
+                )
+                
+                is_oh[e] = is_OH_element
+                is_bc[e] = is_BC_element
+            is_emis = ~np.equal(is_oh | is_bc, True)
+            
+            # fill pert base array with values
+            pert_base_xch4 = np.full(n_elements, np.nan)
+            pert_base_xch4 = np.where(is_emis, emis_base_xch4, pert_base_xch4)
+            pert_base_xch4 = np.where(is_bc, emis_base_xch4, pert_base_xch4)
```

Comment:
can you put a comment explaining why this logic is here?

Reply:
Added better documentation: https://github.com/geoschem/integrated_methane_inversion/pull/252/commits/07e05a19bd0e48c36442d712feb2567a033b3162